packages/core/src/common/utils.ts

```diff
@@ -104,3 +104,40 @@ export function throttleEvent(target: EventTarget, eventName: string, newEventNa
     target.addEventListener(eventName, func);
     return func;
 };
+
+export interface IThrottledReactEventOptions {
+    preventDefault?: boolean;
+};
+
+/**
+ * Throttle a callback by wrapping it in a `requestAnimationFrame` call. Returns the throttled
+ * function.
+ * @see https://www.html5rocks.com/en/tutorials/speed/animations/
+ */
+export function throttleReactEventCallback(
+    callback: (event: React.SyntheticEvent<any>, ...otherArgs: any[]) => any,
+    options: IThrottledReactEventOptions = {},
+) {
+    let isRunning = false;
+    const func = (event2: React.SyntheticEvent<any>, ...otherArgs2: any[]) => {
+        // need to preventDefault synchronously and without regard to the rAF timeline,
+        // if we want it to work
+        if (options.preventDefault) {
+            event2.preventDefault();
+        }
+
+        if (isRunning) {
+            return;
+        }
+        isRunning = true;
+
+        // prevent React from reclaiming the event object before we reference it
+        event2.persist();
+
+        requestAnimationFrame(() => {
+            callback(event2, ...otherArgs2);
+            isRunning = false;
+        });
+    };
+    return func;
+}
```

Comment:
hmmm would be cool for the two throttle util functions here to share _some_ (if not most) of their code...